packages/cli/src/databases/dsl/Indices.ts

```diff
@@ -19,9 +19,18 @@ abstract class IndexOperation extends LazyPromise<void> {
 		protected tablePrefix: string,
 		queryRunner: QueryRunner,
 		protected customIndexName?: string,
+		protected skipIfMissing = false,
```

Comment:
this should only be on the `DropIndex` class

```typescript
export class DropIndex extends IndexOperation {
	constructor(
		tableName: string,
		columnNames: string[],
		tablePrefix: string,
		queryRunner: QueryRunner,
		customIndexName?: string,
		protected skipIfMissing = false,
	) {
		super(tableName, columnNames, tablePrefix, queryRunner, customIndexName);
	}

	async execute(queryRunner: QueryRunner) {
		return await queryRunner
			.dropIndex(this.fullTableName, this.customIndexName ?? this.fullIndexName)
			.catch((error) => {
				if (error instanceof Error && error.message.includes('not found') && this.skipIfMissing) {
					return;
				}
				throw error;
			});
	}
}
```

Reply:
could we somehow use the `IF EXISTS` parameter for DBs that support it?

Reply:
We could call `queryRunner.query` with handcrafted SQL and then fall back to ignoring the error in a catch clause if `IF EXISTS` is unsupported, but I think the current approach is simpler. Let me know otherwise!